salt/_modules/metalk8s_package_manager_yum.py

```diff
@@ -198,6 +198,8 @@ def check_pkg_availability(pkgs_info):
         try:
             ybase.install(name=name, version=version, release=release)
         except yum.Errors.InstallError as exc:
+            ybase.closeRpmDB()
```

Comment:
Not too fond of doing it this way, because there's now one `close` path in an `except` and a later `finally`. There are better ways to do error handling with resource releasing.

Reply:
implementing a python context and using `with` will close the object every time I guess